### PR TITLE
Connection Rework Phase One

### PR DESCRIPTION
**Pull request by Windchild292**
This is the first phase of the connection rework, split off of #3450.  This PR splits off the ConnectionEventType and PacketCommand Enums, migrates the packet setup to varargs, and creates the events setup.

Finally, this likely has bugs that I've missed in testing. I've tried to test it, but there's a huge chance I've missed something... but we need these changes to fix bugs in the core connection setup without making it too hard to work with.

### CODE REVIEW

File: megamek/src/megamek/common/net/enums/PacketCommand.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        return this == CFR_DOMINO_EFFECT;
+    }
+
+    public boolean isCFRTagTarget() {
+        return this == CFR_TAG_TARGET;
+    }
+
+    public boolean isCFR() {
+        return isSendingBoard() || isCFRHiddenPBS() || isCFRTeleguidedTarget() || isCFRAMSAssign()
+                || isCFRAPDSAssign() || isCFRDominoEffect() || isCFRTagTarget();
```
Comment (SJuliez) [inline megamek/src/megamek/common/net/enums/PacketCommand.java:159]: Sure that sendingBoard is a CFR?

File: megamek/src/megamek/server/Server.java [hunk 2]
```diff
@@ -30428,9 +30433,13 @@ private void receiveGameOptionsAux(Packet packet, int connId) {
...
-        for (AbstractConnection conn : connections) {
-            send(conn.getId(), new Packet(Packet.COMMAND_GAME_VICTORY_EVENT));
+        if (connections == null) {
+            return;
         }
+
+        connections.stream()
+                .filter(Objects::nonNull)
+                .forEach(connection -> connection.send(new Packet(PacketCommand.GAME_VICTORY_EVENT)));
     }
```
Comment (SJuliez) [inline megamek/src/megamek/server/Server.java:30443]: Just to make sure: It is intentional that this is no longer Server::send(...), instead it is now Connection::send? I looked at it quickly and it looked to me like it ends up being the same. But just making sure.

**Review by SJuliez — APPROVED** [review]
OK for what it's worth, this looks good to me. Two places where I left a question.

File: megamek/src/megamek/common/net/enums/PacketCommand.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/common/net/enums/PacketCommand.java:159]: Nice catch, fixed.

File: megamek/src/megamek/server/Server.java (hunk 2 above)
Comment (Windchild292) [reply megamek/src/megamek/server/Server.java:30443]: It's identical, and easier to read this way. I've further changed it to use Server::send(Packet), which handles null checks and send to all available connections.

**Pull request merged**